WeakAuras/Prototypes.lua

```diff
@@ -6602,21 +6602,162 @@ Private.event_prototypes = {
+        trigger.use_inverse and "true" or "false",
+        trigger.use_interruptSchool and "true" or "false",
+        trigger.interruptSchool or 0
+      )
+      return ret
+    end,
     args = {
       {
-        name = "controlled",
-        display = L["Crowd Controlled"],
-        type = "tristate",
-        init = "not HasFullControl()"
-      }
+        name = "controlType",
+        display = L["Specific CC Type"],
+        type = "select",
+        values = "loss_of_control_Types",
+        conditionType = "select",
+        test = "true",
+        default = "STUN",
+      },
+      {
+        name = "interruptSchool",
+        display = L["Interrupt School"],
+        type = "select",
+        values = "main_spell_schools",
+        conditionType = "select",
+        default = 1,
+        conditionType = "select",
+        test = "true",
+        enable = function(trigger) return trigger.controlType == "SCHOOL_INTERRUPT" end,
+      },
+      {
+        name = "inverse",
+        display = L["Inverse"],
+        type = "toggle",
+        test = "true",
+      },
+      {
+        name = "name",
+        display = L["CC Name"],
+        hidden = true,
+        conditionType = "string",
+        init = "controlText",
+        store = true,
+        test = "true",
+      },
+      {
+        name = "spellName",
+        display = L["CC Spell Name"],
+        hidden = true,
+        conditionType = "string",
+        init = "spellName",
+        store = true,
+        test = "true",
+      },
+      {
+        name = "locType",
+        display = L["CC Type"],
+        hidden = true,
+        conditionType = "string",
+        init = "locType",
+        store = true,
+        test = "true",
+      },
+      {
+        name = "spellId",
+        display = L["CC Spell Id"],
+        hidden = true,
+        conditionType = "number",
+        init = "spellId",
+        store = true,
+        test = "true",
+      },
+      {
+        name = "lockoutSchool",
+        display = L["Interrupted School"],
+        hidden = true,
+        conditionType = "string",
+        init = "lockoutSchool and lockoutSchool > 0 and GetSchoolString(lockoutSchool) or nil",
+        store = true,
+        test = "true",
+      },
+      {
+        name = "icon",
+        hidden = true,
+        init = "icon",
+        store = true,
+        test = "true",
+      },
+      {
+        name = "duration",
+        hidden = true,
+        init = "duration",
+        store = true,
+        test = "true",
+      },
+      {
+        name = "expirationTime",
+        init = "expirationTime",
+        hidden = true,
+        store = true,
+        test = "true",
+      },
+      {
+        name = "progressType",
+        hidden = true,
+        init = "'timed'",
+        store = true,
+        test = "true",
+      },
+      {
+        hidden = true,
+        test = "(inverse and not show) or (not inverse and show)",
```

Comment:
You could write that as "inverse ~= show", which is a bit less readable. Though it's fine in this case.